### ISSUE TITLE
Inconsistencies in `__contains__` implementation across complex types

### ISSUE THREAD

**Issue by ffl096** [pyt-team/TopoNetX#273]
The function `__contains__` is not consistently implemented across different complex types:

- For cell and combinatorial complexes, it checks whether a node exists.
- For simplicial and path complexes, it checks whether an atom (nodes but also higher-order) exists.

The `Complex` ABC documents that `__contains__` checks whether the complex contains an item.

**Comment by mhajij** [comment]
@ffl096 there is an issue here though. Iter iterates only over nodes only now consistent with NetworkX. So it would make more sense to make __getitem__  also works same way? namely it will only check if a node is in the complex. If one wants to check if an item is in the complex then one should check in CC.cells or PC.paths or SC.simplices, also consistent with NetworkX, what do you think?

**Comment by ffl096** [comment]
I agree on that, `__iter__`, `__contains__` and so on should be consistent in that a complex does not "contain" something that is not iterated over ect. We have to update the documentation of `Complex.__contains__` for that.
